### PR TITLE
Kernel Automation Build Script

### PR DESCRIPTION
**Pull request by RuffaloLavoisier**
Kernel Automation Build Script

I produced Raspberry Pie Build Script with a shell script. 

1. Process calculation for make speed is possible.
2. Optional build is possible by entering the raspberry pie board model as a parameter.

### CODE REVIEW

**Comment by pelwell** [comment]
This script probably works really well for you, and I applaud your desire to help others, but its limited to cross-compilation of a 32-bit kernel on a Linux-based host, with the destination OS in a card reader, and we don't need more software to support.

**Comment by RuffaloLavoisier** [comment]
> This script probably works really well for you, and I applaud your desire to help others, but its limited to cross-compilation of a 32-bit kernel on a Linux-based host, with the destination OS in a card reader, and we don't need more software to support.

Thank you

We don't need a script like this? Even if I make improvements?

**Comment by pelwell** [comment]
I prefer to document how to use the standard kernel makefiles (https://www.raspberrypi.org/documentation/linux/kernel/building.md), then if something goes wrong people have more of a chance of knowing what was happening at the time. But other opinions are available - does anybody want to speak up in support of including this script or a derivative thereof?

**Comment by RuffaloLavoisier** [comment]
> I prefer to document how to use the standard kernel makefiles (https://www.raspberrypi.org/documentation/linux/kernel/building.md), then if something goes wrong people have more of a chance of knowing what was happening at the time. But other opinions are available - does anybody want to speak up in support of including this script or a derivative thereof?

Thank you. I'll help to contribute to the storage by referring to it.

File: Build.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+sudo cp /media/$USER/boot/$KERNEL.img /media/$USER/boot/$KERNEL-backup.img
+sudo cp arch/arm/boot/zImage /media/$USER/boot/$KERNEL.img
+sudo cp arch/arm/boot/dts/*.dtb /media/$USER/boot
+sudo cp arch/arm/boot/dts/overlays/*.dtb* /media/$USER/boot/overlays/
+sudo cp arch/arm/boot/dts/overlays/README /media/$USER/boot/overlays/
+
+echo "Finish - $(date)"
+}
+
+build_Script $1
```
Comment (RuffaloLavoisier) [inline Build.sh]: Receive shell script parameters (board species)

File: Build.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
+function build_Script(){
+bit=$(getconf LONG_BIT)
```
Comment (RuffaloLavoisier) [inline Build.sh]: Check the bits on the host computer.

File: Build.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	KERNEL=kernel7 # pi 3
+	make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- bcm2709_defconfig # pi 3
+	echo "Pi 3 Board Config"
+elif [ $1 = "4" ]; then
+	KERNEL=kernel7l # pi 4
+	make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- bcm2711_defconfig # pi 4
+	echo "Pi 4 Board Config"
+fi
+
+num=$(expr $(grep ^processor /proc/cpuinfo | wc -l) \* 2)
```
Comment (RuffaloLavoisier) [inline Build.sh]: Increase the make speed according to the number of computer processes.

**Review by RuffaloLavoisier — COMMENTED** [review]
Add Shell Script code review comment.

**Comment by RuffaloLavoisier** [comment]
@pelwell Hi, 60 days have passed since Pull request. There is no special reaction, so I will close the Pull Request.

**Pull request closed without merging**